frontend/src/pages/NewModel.tsx

```diff
@@ -1,96 +1,235 @@
                         <button
                             type="button"
-                            className="inline-flex w-full justify-center rounded-md bg-primary-400 hover:bg-primary-600 px-3 py-2 text-sm font-semibold text-dark-400 shadow-sm sm:ml-3 sm:w-auto"
-                            // disabled={!!name || !!modelPath || !!modelType}
+                            className={`${selectedFile ?
+                                " cursor-pointer inline-flex w-full justify-center rounded-md bg-primary-200 hover:bg-primary-400 px-3 py-2 text-sm font-semibold text-dark-400 shadow-sm sm:ml-3 sm:w-auto" :
+                                " cursor-not-allowed inline-flex w-full justify-center rounded-md bg-dark-200 px-3 py-2 text-sm font-semibold text-gray-400/70 shadow-sm sm:ml-3 sm:w-auto"}`}
+                            disabled={!submitAllowed}
                             onClick={() => {
-                                registerHandler({
-                                    name: name,
-                                    version: version || undefined,
-                                    model_type: modelType,
-                                    model_params: {
-                                        model_path: modelPath,
-                                    }
-                                })
+                                if (selectedModel && selectedFile) {
+                                    importModelMutation({
+                                        type: 'locatorv1/hf',
+                                        repo: selectedModel,
+                                        file: selectedFile,
+                                    })
+                                        .unwrap()
+                                        .then(importJobId => {
+                                            setImportJob(importJobId);
+                                        })
+                                        .catch(error => {
+                                            // What to do about the error, update that we have a failed to import thing?
+                                            setImportJob(undefined);
+                                            setImportError(error);
+                                        });
+                                }
                             }}>
                             Register New Model
                         </button>
                     </div>
-                </form>
+                    {importJob &&
+                        (<div className="mt-5">
+                            <p className="text-gray-400/80">{importIsLoading && "loading..."} {importData ? JSON.stringify(importData) : ""}</p>
+                        </div>)
+                    }
+                </>
+            )}
+        </form>
+    )
+}
+
+export default function NewModel() {
+    const [selection, setSelection] = useState("none");
```

Comment:
Can we make this consistent with the other things, i.e.

```
const [selection, setSelection] = useState<string | undefined>();
```

Reply:
Haha see my response to your prior comment, aligned!

Reply:
Addressed!